Sources/xcodeproj/Objects/Project/PBXObjectReference.swift

```diff
@@ -103,17 +99,49 @@ class PBXObjectReference: NSObject, Comparable, NSCopying {
         return lhs.value < rhs.value
     }
 
+    /// Sets the object so it can be retrieved quickly again later
+    ///
+    /// - Parameter object: The object
+    func setObject(_ object: PBXObject) {
+        objectRef = object
+    }
+
+    /// Returns the object the reference is referfing to.
+    ///
+    /// - Returns: object the reference is referring to. Returns nil if the objects property has been released or the reference doesn't exist
+    func object<T: PBXObject>() -> T? {
+        return try? objectOrThrow()
+    }
+
     /// Returns the object the reference is referfing to.
     ///
     /// - Returns: object the reference is referring to.
     /// - Throws: an errof it the objects property has been released or the reference doesn't exist.
-    func object<T>() throws -> T {
+    func objectOrThrow<T: PBXObject>() throws -> T {
```

Comment:
What about calling this method `materialize`? If it sounds too ambiguous we can leave the one that you chose, it's fine.

Reply:
I've seen the term being used in reactive libraries where a type wraps another type. I think it's a similar notion here, but in this case, we wrap it through a reference. Anyway, feel free to make the call here. This method not internal so we can rename it later if we are not quite happy with how it reads in the codebase.